xarray/tests/test_backends.py

```diff
@@ -138,96 +138,96 @@ def open_example_mfdataset(names, *args, **kwargs) -> Dataset:
     )
 
 
-def create_masked_and_scaled_data() -> Dataset:
-    x = np.array([np.nan, np.nan, 10, 10.1, 10.2], dtype=np.float32)
+def create_masked_and_scaled_data(dtype=np.float32) -> Dataset:
```

Comment:
```suggestion
def create_masked_and_scaled_data(dtype: np.typing.DTypeLike = np.float32) -> Dataset:
```

Can do this in other places too.